ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/TransitionView.kt

```diff
@@ -0,0 +1,137 @@
+package com.revenuecat.purchases.ui.revenuecatui.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.core.Easing
+import androidx.compose.animation.core.FastOutLinearInEasing
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.LinearOutSlowInEasing
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.scaleIn
+import androidx.compose.animation.slideIn
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.layout
+import androidx.compose.ui.unit.IntOffset
+import com.revenuecat.purchases.paywalls.components.PaywallAnimation
+import com.revenuecat.purchases.paywalls.components.PaywallAnimation.AnimationType
+import com.revenuecat.purchases.paywalls.components.PaywallTransition
+import com.revenuecat.purchases.paywalls.components.PaywallTransition.TransitionType.Fade
+import com.revenuecat.purchases.paywalls.components.PaywallTransition.TransitionType.FadeAndScale
+import com.revenuecat.purchases.paywalls.components.PaywallTransition.TransitionType.Scale
+import com.revenuecat.purchases.paywalls.components.PaywallTransition.TransitionType.Slide
+
+@Suppress("ModifierMissing")
+@Composable
+internal fun TransitionView(transition: PaywallTransition?, content: @Composable () -> Unit) {
+    if (transition == null) {
+        content()
+    } else {
+        if (transition.displacementStrategy == PaywallTransition.DisplacementStrategy.GREEDY) {
+            Box {
+                Box(modifier = Modifier.hidden()) {
+                    content()
+                }
+
+                transition.AnimatedVisibility { content() }
+            }
+        } else {
+            transition.AnimatedVisibility { content() }
+        }
+    }
+}
+
+private fun Modifier.hidden(): Modifier = this.layout { measurable, constraints ->
```

Comment:
Hmm I was thinking whether it would make more sense to just add the content again with an alpha of 0, but then we would be rendering the content twice. I think this seems like a smart solution! We should just test to make sure it behaves as expected, but seems right 👍